pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -278,16 +278,20 @@ func CollectResourceRequestsAndLimits(vms []*k6tv1.VirtualMachine) []operatormet
 	var results []operatormetrics.CollectorResult
 
 	for _, vm := range vms {
+		// Apply any instance type and preference to a copy of the VM before proceeding
+		vmCopy := vm.DeepCopy()
```

Comment:
@machadovilaca This is going to be a memory crunching operation on a large scale clusters. Can we somehow calculate it w/o doing the real VM apply? The effort worth it, even if it requires additional functionality on the instance type package

Reply:
@lyarwood what do you think?

Reply:
@enp0s3 @machadovilaca I personally think the limited computational hit here with the `DeepCopy` is worth it to keep the below code simple and unified. Each copy is scoped locally within the loop so should be free'd by the GC right? This should be easy enough to profile and prove one way or another, I can take this on either in a documented follow up issue or before this lands.